src/app/inventory/store/stats.ts

```diff
@@ -116,30 +117,27 @@ export function buildStats(
 
   // We only use the raw "investment" stats to calculate all item stats.
   let investmentStats = buildInvestmentStats(itemDef, defs, statGroup, statDisplays) || [];
-  let investmentStatsByHash = _.keyBy(investmentStats, (s) => s.statHash);
 
   // Include the contributions from perks and mods
-  if (createdItem.sockets?.allSockets.length) {
-    enhanceStatsWithPlugs(
+  enhanceStatsWithPlugs(itemDef, investmentStats, createdItem, defs, statGroup, statDisplays);
+
+  // For Armor, we always replace the previous stats with live stats, even if they were already created
+  if (
+    (!investmentStats.length || createdItem.bucket.inArmor) &&
+    stats?.[createdItem.id] &&
+    createdItem.type !== 'ClassItem'
+  ) {
+    // TODO: build a version of enhanceStatsWithPlugs that only calculates plug values
```

Comment:
That's a good idea!

Reply:
I would love to take credit for your todo's but I just can't haha.